File: Patents/2025 Patents/Provisional Filing/T-Mobile Content Aware Widget Arbitration/WidgetArbitration/Dev/WidgetArbitrationManager.java
package com.tmobile.widgetarbitration;

import java.util.Map;

public class WidgetArbitrationManager {

    public static class WidgetRefreshDecision {
        public final boolean shouldRefresh;
        public final String reason;

        public WidgetRefreshDecision(boolean shouldRefresh, String reason) {
            this.shouldRefresh = shouldRefresh;
            this.reason = reason;
        }
    }

    public static boolean isOverrideForced(String appPackage) {
        return appPackage.contains("sponsored");
    }

    public static double scoreEngagement(String widgetId, Map<String, Object> metadata) {
        // Placeholder ML logic
        return 0.85;
    }

    public static WidgetRefreshDecision shouldRefreshWidget(String widgetId, Map<String, Object> metadata, String appPackage) {
        if (isOverrideForced(appPackage)) {
            return new WidgetRefreshDecision(true, "Override by policy");
        }
        double score = scoreEngagement(widgetId, metadata);
        if (score >= 0.6) {
            return new WidgetRefreshDecision(true, "Engagement score passed");
        } else {
            return new WidgetRefreshDecision(false, "Suppressed due to low engagement");
        }
    }
}

File: Patents/2025 Patents/Provisional Filing/T-Mobile Content Aware Widget Arbitration/WidgetArbitration/Dev/WidgetArbitrationManager.kt
package com.tmobile.widgetarbitration

data class WidgetRefreshDecision(val shouldRefresh: Boolean, val reason: String)

object EngagementScorer {
    fun scoreEngagement(widgetId: String, metadata: Map<String, Any>): Double {
        // Simulate ML logic
        return 0.85 // Placeholder
    }
}

object OverridePolicyHandler {
    fun isOverrideForced(appPackage: String): Boolean {
        return appPackage.contains("sponsored")
    }
}

object WidgetArbitrationManager {
    fun shouldRefreshWidget(widgetId: String, metadata: Map<String, Any>, appPackage: String): WidgetRefreshDecision {
        if (OverridePolicyHandler.isOverrideForced(appPackage)) {
            return WidgetRefreshDecision(true, "Override by policy")
        }
        val score = EngagementScorer.scoreEngagement(widgetId, metadata)
        return if (score >= 0.6) {
            WidgetRefreshDecision(true, "Engagement score passed")
        } else {
            WidgetRefreshDecision(false, "Suppressed due to low engagement")
        }
    }
}